README.md

```diff
@@ -313,6 +313,27 @@ To add Prometheus exposition to an existing HTTP server, see the `MetricsHandler
 which provides a `BaseHTTPRequestHandler`. It also serves as a simple example of how
 to write a custom endpoint.
 
+By default, the prometheus client will accept only HTTP requests from Prometheus.
+To enable HTTPS, `certfile` and `keyfile` need to be provided. The certificate is
+presented to Prometheus as a server certificate during the TLS handshake, and
+the private key in the key file must belong to the public key in the certificate.
+
+When HTTPS is enabled, by default mutual TLS (mTLS) is enforced (i.e. Prometheus is
```

Comment:
This is incorrect, it is only when a CA file is also present

Reply:
My understanding is that when the `cafile` parameter is not specified (i.e. `None`), the default CA certificate chain is used as described in https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_default_certs.

The enforcement of mTLS is done by setting `ssl_cxt.verify_mode = ssl.CERT_REQUIRED`, and this is done regardless of `cafile`(or `capath`) being set. The only conditions are that HTTPS was selected and `insecure_skip_verify=False`.

Reply:
@roidelapluie I would like to keep this as similar to other exporters as possible, do they require mTLS by default?

Reply:
@csmarchbanks @roidelapluie I'm fine with any way to enable mTLS. Let's keep it consistent with the other exporters then.

Reply:
I am happy with that proposal, and agree that supporting the web config file is outside the scope of this PR, and possibly this package. Let's keep it as simple as possible.

Reply:
@karezachen Hi Kareza. Since Chris has agreed to the proposal, could you please update the PR to implement it:
- Rename the `insecure_skip_verify` parameter to `client_auth_required`.
- This turns around its semantics, so keep the default as `False` but negate any use of it in the code.

Reply:
Can we just require client auth if a client_cafile is specified?

Reply:
@andy-maier Hi Andy. I have made the following updates:
- Rename the insecure_skip_verify parameter to client_auth_required.
- Update `_get_ssl_ctx` func, default not to set `ssl_ctx.verify_mode=ssl.CERT_REQUIRED`.
- Update the description of default in README.md.
---
And I have updated the Demo in the description of this PR based on the latest code.

Reply:
> Can we just require client auth if a client_cafile is specified?

@andy-maier Andy, I don't quite understand this problem. Can you help me? Thanks.